internal/cli/cli.go

```diff
@@ -122,3 +124,125 @@ func (opts *deleteOpts) FailMessage() string {
+	notificationToken               string // notificationsApiToken, notificationsFlowdockApiToken
+	notificationChannelName         string
+	apiKey                          string // notificationsDatadogApiKey, notificationsOpsGenieApiKey, notificationsVictorOpsApiKey
+	notificationEmailAddress        string
+	notificationFlowName            string
+	notificationMobileNumber        string
+	notificationRegion              string // notificationsOpsGenieRegion, notificationsDatadogRegion
+	notificationOrgName             string
+	notificationServiceKey          string
+	notificationTeamID              string
+	notificationType                string
+	notificationUsername            string
+	notificationVictorOpsRoutingKey string
+	notificationDelayMin            int
+	notificationIntervalMin         int
+	notificationSmsEnabled          bool
+	enabled                         bool
+	notificationEmailEnabled        bool
+	metricThresholdThreshold        float64
+}
+
+func (opts *AtlasAlertsConfig) buildAlertConfiguration() *atlas.AlertConfiguration {
+
+	out := new(atlas.AlertConfiguration)
+
+	out.GroupID = opts.ProjectID()
+	out.EventTypeName = strings.ToUpper(opts.event)
+	out.Enabled = &opts.enabled
+
+	if opts.matcherFieldName != "" {
+		out.Matchers = []atlas.Matcher{*newMatcher(opts)}
+	}
+
+	if opts.metricThresholdMetricName != "" {
+		out.MetricThreshold = newMetricThreshold(opts)
+	}
+
+	out.Notifications = []atlas.Notification{*newNotification(opts)}
+
+	return out
+}
+
+func newNotification(opts *AtlasAlertsConfig) *atlas.Notification {
+
+	out := new(atlas.Notification)
+	out.TypeName = strings.ToUpper(opts.notificationType)
+	out.DelayMin = &opts.notificationDelayMin
+	out.IntervalMin = opts.notificationIntervalMin
+	out.TeamID = opts.notificationTeamID
+	out.Username = opts.notificationUsername
+	out.ChannelName = opts.notificationChannelName
+
+	switch out.TypeName {
+
+	case victor:
+		out.VictorOpsAPIKey = opts.apiKey
+		out.VictorOpsRoutingKey = opts.notificationVictorOpsRoutingKey
+
+	case slack:
+		out.VictorOpsAPIKey = opts.apiKey
+		out.VictorOpsRoutingKey = opts.notificationVictorOpsRoutingKey
+		out.APIToken = opts.notificationToken
+
+	case datadog:
+		out.DatadogAPIKey = opts.apiKey
+		out.DatadogRegion = strings.ToUpper(opts.notificationRegion)
+
+	case email:
+		out.EmailAddress = opts.notificationEmailAddress
+
+	case flowdock:
+		out.FlowdockAPIToken = opts.notificationToken
+		out.FlowName = opts.notificationFlowName
+		out.OrgName = opts.notificationOrgName
+
+	case sms:
+		out.MobileNumber = opts.notificationMobileNumber
+
+	case group, user, org:
+		out.SMSEnabled = &opts.notificationSmsEnabled
+		out.EmailEnabled = &opts.notificationEmailEnabled
+
+	case ops:
+		out.OpsGenieAPIKey = opts.apiKey
+		out.OpsGenieRegion = opts.notificationRegion
+
+	case pager:
+		out.ServiceKey = opts.notificationServiceKey
+
+	}
+
+	return out
+}
+
+func newMetricThreshold(opts *AtlasAlertsConfig) *atlas.MetricThreshold {
```

Comment:
You can make this a function of `AtlasAlertsConfig` (also remember to rename it as above
```suggestion
func (opts *AtlasAlertsConfig) newMetricThreshold *atlas.MetricThreshold {
```